apps/ex_shop/web/models/order.ex

```diff
@@ -40,6 +47,24 @@ defmodule ExShop.Order do
   def in_cart_state?(%Order{state: "cart"}), do: true
   def in_cart_state?(%Order{state: _}), do: false
 
+  def shipping_address(%Order{} = order) do
+    [address1, address2] = order |> ExShop.Repo.preload([:addresses]) |> Map.get(:addresses)
+    if address1.address_type == "shipping" do
```

Comment:
Can we move to address / order and exploit pattern_match to make it more declarative like get_shipping_address ?

not sure on a sure alternative :(

- one thought is on the lines of zone belonging to countries and states
- exposing functions on address pattern matching on address_type and returning appropriate query to be used in order ??